go/worker/sandbox/containerd.go

```diff
@@ -0,0 +1,162 @@
+package sandbox
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"log/slog"
+        "io/ioutil"
+        "path/filepath"
+
+	"github.com/containerd/containerd"
+        
+	"github.com/open-lambda/open-lambda/go/common"
+	"github.com/open-lambda/open-lambda/go/worker/sandbox/containerdutil"
+)
+
+
+type ContainerdContainer struct {
+	// core containerd resources
+	id        	string
+	container 	containerd.Container  // containerd container handle
+	task      	containerd.Task       // Running process inside the container
+	client		*containerd.Client
+	ctx			context.Context
+	scratchDir	string
+
+	execProcess	containerd.Process  
+
+	// state tracking
+	destroyed	bool        // tracks if WE destroyed it, not just if it's gone
+	isPaused	bool        // cached pause state to avoid API calls
+	// Lambda execution resources
+	meta		*SandboxMeta
+	rtType		common.RuntimeType
+	httpClient	*http.Client
+}
+
+func (c *ContainerdContainer) ID() string {
+	return c.id
+}
+
+func (c *ContainerdContainer) Destroy(reason string) {
+	if c.destroyed {
```

Comment:
Do we need this sort of check, or does the safeSandbox layer prevent such duplicate calls?  I'm not sure, so please look into it.